start.py

```diff
@@ -217,8 +224,10 @@ def cm_server_not_dead(primary_node):
                       files=['/var/lib/cloudera-scm-agent/uuid',
                              '/dfs*/dn/current/*'])
 
+    centos_major_version = primary_node.execute("rpm -q --queryformat '%{VERSION}' centos-release", quiet=True)
+    is_centos_major_version_7 = centos_major_version.output == '7'
     logger.info('Restarting Cloudera Manager agents ...')
-    _restart_cm_agents(nodes)
+    _restart_cm_agents(nodes, is_centos_major_version_7)
```

Comment:
Just as a thought, perhaps it'd be better to keep the function signature as it used to be (e.g. `_restart_cm_agents(nodes)` and, in the body of the function, check there for whether a given node has CentOS 7 installed. That way, we don't end up needing to keep adding more and more args here as we expand OS support. Plus, the body of the function has access to the nodes, so might as well hide this implementation detail there as opposed to bubbling it up in the signature.

Reply:
Done.